src/Microsoft.AspNet.Mvc.Core/Builder/MvcApplicationBuilderExtensions.cs

```diff
@@ -77,7 +79,13 @@ public static IApplicationBuilder UseMvcWithDefaultRoute(this IApplicationBuilde
 
             // Verify if AddMvc was done before calling UseMvc
             // We use the MvcMarkerService to make sure if all the services were added.
-            MvcServicesHelper.ThrowIfMvcNotRegistered(app.ApplicationServices);
+            if (app.ApplicationServices.GetService(typeof(MvcMarkerService)) == null)
+            {
+                throw new InvalidOperationException(Resources.FormatUnableToFindServices(
+                    nameof(IServiceCollection),
+                    "AddMvc",
```

Comment:
Use `nameof(MvcServiceCollectionExtensions.AddMvc)`. Since this is in the Core repo, should also mention `nameof(MvcCoreServiceCollectionExtensions.AddMvcCore)`.